### ISSUE TITLE
regression: failed to resolve proc macro definition

### ISSUE THREAD

**Issue by Mark-Simulacrum** [rust-lang/rust#106337]
* https://crater-reports.s3.amazonaws.com/beta-1.67-2.4/beta-2022-12-29/reg/robin-0.3.0/log.txt

The prelude module from which the macro should be imported is defined like this:

```
    pub use serde::Serialize;
    pub use job::{Args, Job, JobName, JobResult, PerformJob};
    pub use error::RobinResult;
    pub use connection::{establish, LookupJob, WorkerConnection};
    pub use worker::boot;
    pub use config::Config;
    pub use robin_derives::*;
    pub use macros::*;
```

I suspect that the glob import here isn't properly being treated as introducing Job, but without a bisection/further debugging hard to say.

cc @petrochenkov

**Comment by Mark-Simulacrum** [comment]
https://crater-reports.s3.amazonaws.com/beta-1.67-2.4/beta-2022-12-29/reg/box_shorthand-0.1.0/log.txt might be related, not sure.

**Comment by apiraino** [comment]
`cargo +beta-2022-12-29 test` to reproduce for crate `box_shorthand`. The bisection of both points at commit ab5a2bc7316012ee9b2a4a4f3821673f2677f3d5, which is a rollup. Probably @Mark-Simulacrum you meant #103760 from that rollup, correct?

I cannot reproduce the error on the "robin" crate, I only get the warnings. Are these warnings fine?
I'm running:
```
$ cargo +beta-2022-12-29 --version
cargo 1.67.0-beta.5 (f6e737b1e 2022-12-02)`

$ CARGO_INCREMENTAL=0 RUSTFLAGS=--cap-lints=warn \
    cargo +beta-2022-12-29 test --no-run --frozen
```

**Comment by apiraino** [comment]
WG-prioritization assigning priority ([Zulip discussion](https://rust-lang.zulipchat.com/#narrow/stream/245100-t-compiler.2Fwg-prioritization.2Falerts/topic/.23106337.20regression.3A.20failed.20to.20resolve.20proc.20macro.20definition)).

@rustbot label -I-prioritize +P-critical

**Comment by pnkfelix** [comment]
@rustbot label: +P-high -P-critical

(downgrading priority as PR #84022 has been reverted via #107133  (and in beta via #107164 )

**Comment by apiraino** [comment]
can this be closed since the backport of #107133?
